src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -109,14 +111,29 @@ protected void runTool(final JavaSparkContext ctx) {
         }
 
         final Broadcast<ReferenceMultiSource> referenceMultiSourceBroadcast = ctx.broadcast(getReference());
+        final Broadcast<SAMSequenceDictionary> sequenceDictionaryBroadcast = ctx.broadcast(header.getSequenceDictionary());
+        dispatchJobs(contigsByPossibleRawTypes, referenceMultiSourceBroadcast, sequenceDictionaryBroadcast);
+    }
+
+    private void dispatchJobs(final EnumMap<RawTypes, JavaRDD<AlignedContig>> contigsByPossibleRawTypes,
+                              final Broadcast<ReferenceMultiSource> referenceMultiSourceBroadcast,
+                              final Broadcast<SAMSequenceDictionary> sequenceDictionaryBroadcast) {
 
         new InsDelVariantDetector()
-                .inferSvAndWriteVCF(contigsByPossibleRawTypes.get(RawTypes.InsDel), outputDir+"/"+RawTypes.InsDel.name()+".vcf",
-                        referenceMultiSourceBroadcast, discoverStageArgs.fastaReference, localLogger);
+                .inferSvAndWriteVCF(contigsByPossibleRawTypes.get(RawTypes.InsDel), outputDir+"/"+ RawTypes.InsDel.name()+".vcf",
+                        referenceMultiSourceBroadcast, sequenceDictionaryBroadcast, localLogger);
 
         new SimpleStrandSwitchVariantDetector()
-                .inferSvAndWriteVCF(contigsByPossibleRawTypes.get(RawTypes.Inv), outputDir+"/"+RawTypes.Inv.name()+".vcf",
-                        referenceMultiSourceBroadcast, discoverStageArgs.fastaReference, localLogger);
+                .inferSvAndWriteVCF(contigsByPossibleRawTypes.get(RawTypes.Inv), outputDir+"/"+ RawTypes.Inv.name()+".vcf",
+                        referenceMultiSourceBroadcast, sequenceDictionaryBroadcast, localLogger);
+
+        // here contigs supporting cpx having only 2 alignments could be handled easily together with ref block order switch ones
+        final JavaRDD<AlignedContig> diffChrTrans =
+                contigsByPossibleRawTypes.get(RawTypes.Cpx).filter(tig -> tig.alignmentIntervals.size() == 2);
+        new SuspectedTransLocDetector()
+                .inferSvAndWriteVCF(contigsByPossibleRawTypes.get(RawTypes.DispersedDupOrMEI).union(diffChrTrans),
```

Comment:
Why not just put the two classes of alignment in the same bucket to begin with (in the "divertReads" method)? Might be more efficient, and they are likely all the same type of event.

Reply:
There's a branch for this purpose, just not ready yet for a PR...